lib/phobos/producer.rb

```diff
@@ -42,6 +52,30 @@ def async_publish_list(messages)
       def class_producer
         @host_obj.class.producer
       end
+
+      # rubocop:disable Metrics/ParameterLists
+      def normalize_arguments(p_topic = nil, p_payload = nil, p_key = nil,
+                              p_partition_key = nil, p_headers = {},
+                              **kwargs)
+        {}.tap do |args|
+          {
+            topic: p_topic,
+            payload: p_payload,
+            key: p_key,
+            partition_key: p_partition_key,
+            headers: p_headers
+          }.each { |k, v| args[k] = kwargs[k] || v }
+
+          raise MissingRequiredArgumentsError if [:topic, :payload].any? { |k| args[k].nil? }
+
+          kwargs.each do |k, v|
+            next if args.key?(k)
+
+            args[:headers][k] = v
```

Comment:
We need to also print a deprecation message if there are no keyword arguments.

Reply:
@dorner Done ✅ 

I had to change the implementation of `Phobos.deprecate`. It was using `Kernel.caller.first` directly, However because code inside the gem was calling that method, the caller was not the gem's client code but actually the gem itself! So the locations being printed out in the deprecation messages were misleading.